base/sparse/higherorderfns.jl

```diff
@@ -1105,10 +1106,11 @@ promote_spcontainertype(::FunnelToSparseBC, ::FunnelToSparseBC) = PromoteToSpars
 @inline spbroadcast_c!{N}(f, ::Type{AbstractSparseArray}, ::Type{AbstractArray}, C, B, As::Vararg{Any,N}) =
     broadcast_c!(f, Array, Array, C, B, As...)
 
-@inline _sparsifystructured(A::AbstractSparseArray) = A
-@inline _sparsifystructured(M::StructuredMatrix) = SparseMatrixCSC(M)
-@inline _sparsifystructured(M::Matrix) = SparseMatrixCSC(M)
-@inline _sparsifystructured(V::Vector) = SparseVector(V)
+@inline _sparsifystructured(M::AbstractMatrix) = SparseMatrixCSC(M)
```

Comment:
As I mentioned in #20102, I like these changes: These definitions are safer if we at some point broaden the set of type combinations sparse `broadcast[!]` captures. On the other hand, absent that broadening, being explicit about the types sparse `broadcast[!]` handles and what it does with them is also nice. But then being explicit in `_spcontainertype` should be sufficient, so I favor keeping these changes. Thoughts?